### DISCUSSION TITLE
Grouping global searches by unique org/repository

### DISCUSSION THREAD

**Discussion by lsegal** [community/community#9866] [Other Feature Feedback, Questions, & Ideas] [answered]
Most of the time when I am performing searches I am much more interested in the number of results across unique codebases than I am in the individual search. As an example, I might be trying to answer a question like, "how many Ruby codebases use `:nodoc:`?", or "how common is the `acts_as_paranoid` gem?"

Currently code search seems to provide a very greedy per-repository search results, in other words, for a given term it will list every matching file in a given repository, sorted by _repository popularity_.

While this might make sense for one specific type of search, the practical implication of this is that pretty much every global language specific search is currently enumerating the files of the most popular repository of that language. Here's a bunch of random word searches in the Ruby language, all pointing to pages of results almost exclusively from rails:

![image](https://user-images.githubusercontent.com/686/148660132-35495ee3-9676-414b-a2b3-292483587542.png)
![image](https://user-images.githubusercontent.com/686/148660140-3e4f89bf-6a97-4c46-b382-20b043295445.png)
![image](https://user-images.githubusercontent.com/686/148660199-12789b29-48b7-4a41-8f36-9b15a6ca0a8c.png)

It would be great to have global searches group per-repository results, otherwise you are basically just doing a rails repository search whenever you search anything globally for Ruby; and I'm sure this probably applies to other languages with a single extremely popular org/repo.

**Comment by colinwm** [comment, marked as answer]
Hey @lsegal, thanks for the feedback. We've just made some ranking improvements to increase the diversity of resullts. Let me know what you think!

Here's an example of some search results for a query you provided with our new ranking scheme:
![image](https://user-images.githubusercontent.com/78169714/152876244-bac01db3-944f-4ade-b332-e77f2c21dd15.png)

**Reply by SoKamil** [reply]
@colinwm, while this is useful, this doesn't solve the problem entirely. I'd like to search for repos which uses certain library, but have no duplicates. This way I could learn how many usages are there in the wild and be sure that I didn't open same repo twice when looking for usage patterns.

**Comment by azitabh** [comment]
Having this as a feature would be very useful.

**Comment by Ammar-Husain** [comment]
Had this suggestion been implemented? It would be very cool
